src/map/instance.hpp

```diff
@@ -46,58 +51,71 @@ struct s_instance_map {
 	int16 m, src_m;
 };
 
-struct instance_data {
-	unsigned short type; ///< Instance DB ID
-	enum instance_state state; ///< State of instance
-	enum instance_mode mode; ///< Mode of instance
+/// Instance data
+struct s_instance_data {
+	int id; ///< Instance DB ID
+	enum e_instance_state state; ///< State of instance
+	enum e_instance_mode mode; ///< Mode of instance
 	int owner_id; ///< Owner ID of instance
 	unsigned int keep_limit; ///< Life time of instance
 	int keep_timer; ///< Remaining life time of instance
 	unsigned int idle_limit; ///< Idle time of instance
 	int idle_timer; ///< Remaining idle time of instance
 	struct reg_db regs; ///< Instance variables for scripts
-	struct s_instance_map **map; ///< Dynamic array of maps in instance
-	uint8 cnt_map; ///< Number of maps in an instance
+	std::vector<s_instance_map> map; ///< Array of maps in instance
 };
 
-/// Instance DB entry struct
-struct instance_db {
-	unsigned short id; ///< Instance ID
-	StringBuf *name; ///< Instance name
+/// Instance Idle Queue data
+static struct s_instance_wait {
```

Comment:
There is no need to explicitly declare the struct as static.

Reply:
Good catch. I'm not sure why I even changed that.